azurelinuxagent/common/protocol/metadata.py

```diff
@@ -314,8 +323,10 @@ def update_goal_state(self, forced=False, max_retry=3):
             try:
                 self.update_certs()
                 return
-            except:
+            except Exception as e:
                 logger.verbose("Incarnation is out of date. Update goalstate.")
+                msg = u"Exception updating certs: {0}".format(ustr(e))
+                detailed_msg = '{0} {1}'.format(msg, traceback.extract_tb(get_traceback(e)))
```

Comment:
You haven't used the `detailed_msg` or the 'msg` variable - You'll still need to add that.

```suggestion
                logger.warn(msg)
                detailed_msg = '{0} {1}'.format(msg, traceback.extract_tb(get_traceback(e)))
                logger.verbose(detailed_msg)
```